File: ShadowFrameworkLite20/src/shadow/geometry/functions/data/SFSimpleTexCoordGeometryuvData.java
package shadow.geometry.functions.data;

import shadow.geometry.SFSurfaceFunction;
import shadow.geometry.functions.SFSimpleTexCoordGeometryuv;
import shadow.renderer.data.SFDataAsset;
import shadow.system.data.SFNamedParametersObject;
import shadow.system.data.objects.SFFloat;

public class SFSimpleTexCoordGeometryuvData extends SFDataAsset<SFSurfaceFunction>{

	private SFFloat du=new SFFloat(1);
	private SFFloat dv=new SFFloat(1);
	
	public SFSimpleTexCoordGeometryuvData(){
		prepare();
	}
	
	public SFSimpleTexCoordGeometryuvData(float du,float dv){
		prepare();
		setDu(du);
		setDv(dv);
	}

	private void prepare() {
		SFNamedParametersObject parameters=new SFNamedParametersObject();
		parameters.addObject("du", du);
		parameters.addObject("dv", dv);
		setData(parameters);
	}
	
	@Override
	protected SFSurfaceFunction buildResource() {
		return new SFSimpleTexCoordGeometryuv(du.getFloatValue(), dv.getFloatValue());
	}

	public void set(float du,float dv) {
		this.du.setFloatValue(du);
		this.dv.setFloatValue(dv);
	}
	
	public void setDu(float du) {
		this.du.setFloatValue(du);
	}

	public void setDv(float dv) {
		this.dv.setFloatValue(dv);
	}
}

File: trunk/ShadowFrameworkJS/src/shadow/geometry/functions/data/SFSimpleTexCoordGeometryuvData.js


function SFSimpleTexCoordGeometryuvData(){
	
	this.du=new SFFloat(1);
	this.dv=new SFFloat(1);
	
	var parameters=new SFNamedParametersObject();
	parameters.addObject(this.du);
	parameters.addObject(this.dv);
	this.setData(parameters);
}

inherit(SFSimpleTexCoordGeometryuvData,SFDataAsset);

SFSimpleTexCoordGeometryuvData.prototype["buildResource"]=function(){
	return new SFSimpleTexCoordGeometryuv(this.du.getFloatValue(), this.dv.getFloatValue());
};

SFSimpleTexCoordGeometryuvData.prototype["generateNewDatasetInstance"]=function(){
	return new SFSimpleTexCoordGeometryuvData();
};	
	
